pkg/envoy/embedded_envoy.go

```diff
@@ -137,7 +137,7 @@ func startEmbeddedEnvoy(runDir, logPath string, baseID uint64) (*EmbeddedEnvoy,
 
 		for {
 			logLevel := logging.GetLevel(logging.DefaultLogger)
-			cmd := exec.Command(ciliumEnvoy, "-l", mapLogLevel(logLevel), "-c", bootstrapPath, "--base-id", strconv.FormatUint(baseID, 10), "--log-format", logFormat)
+			cmd := exec.Command(ciliumEnvoyStarter, "-l", mapLogLevel(logLevel), "-c", bootstrapPath, "--base-id", strconv.FormatUint(baseID, 10), "--log-format", logFormat)
```

Comment:
is this one still using cilium-envoy-starter?

Reply:
ah got it, then version check code is not part of this PR

Reply:
well it's impacted by this PR. but since the variable name stays the same (only it's value points to the envoy binary instead of the starter one), we don't see any changes there.